bson/bsonrw/extjson_parser.go

```diff
@@ -273,6 +283,61 @@ func (ejp *extJSONParser) readValue(t bsontype.Type) (*extJSONValue, error) {
 
 		ejp.advanceState()
 		if t == bsontype.Binary && ejp.s == jpsSawValue {
+			// convert relaxed $uuid format
+			if ejp.relaxedUUID {
```

Comment:
Do we need to store this state on `extJSONParser`? Perhaps we could check `ejp.k` on on L278.  Right now `extJSONParser` does not track the type.

Reply:
Hmm it might be possible to do it without the new `relaxedUUID` state, but the existing `ejp.k` check on L126 also forces the type to be `bsontype.Binary` to enter the if check above on L275. Otherwise, we'd still be parsing in the `bsontype.String` case and wouldn't be able to check `ejp.k` here. I'll look into a solution without the extra state/type-tracking.

Reply:
Discussed offline: will keep the `relaxedUUID` state for now. We need it because `peekType()` deals with keys and will recognize "$uuid", while `readValue()` deals with values and will be able to parse the new format. The relaxedUUID state needs to be communicated between them.